apis/tempo/v1alpha1/tempomonolithic_types.go

```diff
@@ -0,0 +1,244 @@
+
+	// ManagementState defines whether this instance is managed by the operator or self-managed
+	//
+	// +kubebuilder:validation:Optional
+	Management ManagementStateType `json:"management,omitempty"`
+
+	// Observability defines observability configuration for the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Observability *MonolithicObservabilitySpec `json:"observability,omitempty"`
+
+	// ExtraConfig defines any extra (overlay) configuration for components
+	//
+	// +kubebuilder:validation:Optional
+	ExtraConfig *MonolithicExtraConfigSpec `json:"extraConfig,omitempty"`
+}
+
+// MonolithicStorageSpec defines the storage for the Tempo deployment.
+type MonolithicStorageSpec struct {
+	// Traces defines the backend storage configuration for traces
+	//
+	// +kubebuilder:validation:Required
+	Traces MonolithicTracesStorageSpec `json:"traces"`
+}
+
+// MonolithicTracesStorageSpec defines the traces storage for the Tempo deployment.
+type MonolithicTracesStorageSpec struct {
+	// Backend defines the backend for storing traces. Default: memory
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=memory
+	Backend MonolithicTracesStorageBackend `json:"backend"`
+
+	// WAL defines the write-ahead logging (WAL) configuration
+	//
+	// +kubebuilder:validation:Optional
+	WAL *MonolithicTracesStorageWALSpec `json:"wal,omitempty"`
+
+	// PV defines the Persistent Volume configuration
+	//
+	// +kubebuilder:validation:Optional
+	PV *MonolithicTracesStoragePVSpec `json:"pv,omitempty"`
+}
+
+// MonolithicTracesStorageBackend defines the backend storage for traces.
+//
+// +kubebuilder:validation:Enum=memory;pv
+type MonolithicTracesStorageBackend string
+
+const (
+	// MonolithicTracesStorageBackendMemory defines storing traces in a tmpfs (in-memory filesystem).
+	MonolithicTracesStorageBackendMemory MonolithicTracesStorageBackend = "memory"
+	// MonolithicTracesStorageBackendPV defines storing traces in a Persistent Volume.
+	MonolithicTracesStorageBackendPV MonolithicTracesStorageBackend = "pv"
+)
+
+// MonolithicTracesStorageWALSpec defines the write-ahead logging (WAL) configuration.
+type MonolithicTracesStorageWALSpec struct {
+	// Size defines the size of the Persistent Volume for storing the WAL. Defaults to 10Gi.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="10Gi"
+	Size resource.Quantity `json:"size"`
+}
+
+// MonolithicTracesStoragePVSpec defines the Persistent Volume configuration.
+type MonolithicTracesStoragePVSpec struct {
+	// Size defines the size of the Persistent Volume for storing the traces. Defaults to 10Gi.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="10Gi"
+	Size resource.Quantity `json:"size"`
+}
+
+// MonolithicIngestionSpec defines the ingestion settings.
+type MonolithicIngestionSpec struct {
+	// OTLP defines the ingestion configuration for OTLP
+	//
+	// +kubebuilder:validation:Optional
+	OTLP *MonolithicIngestionOTLPSpec `json:"otlp,omitempty"`
```

Comment:
It will be the only protocol supported?

Reply:
I want to make the gateway a drop-in feature, so the service ports should not change if gateway is enabled or not. So I can only support protocols which the gateway also supports.
Afaics the gateway only supports otlp/grpc and otlp/http, right?

With the general move to the OTEL SDK, and using the OTEL collector, I think it's fine to only support OTLP.